plugins/modules/cloud/misc/proxmox_kvm.py

```diff
@@ -960,7 +960,13 @@ def create_vm(self, vmid, newid, node, name, memory, cpu, cores, sockets, update
                 kwargs.update(kwargs[k])
                 del kwargs[k]
 
-        # Rename numa_enabled to numa. According the API documentation
+        # Map bool to string, according to API documentation.
+        if kwargs['agent'] is True:
+            kwargs['agent'] = 'enabled=1'
+        elif kwargs['agent'] is False:
+            kwargs['agent'] = 'enabled=0'
```

Comment:
According to https://github.com/ansible-collections/community.general/pull/4027#discussion_r913231029 these are never true since the type is now a string. Is there a way to make the type support either a boolean or a string.

Reply:
Or do I just change these to a string comparison with `'True'` and `'False'`?

Reply:
The comment referred to the fact that that PR was changing the parameter type, same as this PR is doing. Now, IMO, it makes no sense to change it to `str` and then expect `True` or `False`.

Reply:
If you passed in `"yes"`, `"1"`, `"on"`, `"true"`, or several other strings to the old version, they were also interpreted as `True`. So to be backwards compatible just checking for `"True"` is not suficient.